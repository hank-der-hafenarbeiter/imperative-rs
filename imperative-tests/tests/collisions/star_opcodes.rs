use imperative_rs::InstructionSet;

#[derive(InstructionSet)]
enum Star {
    #[opcode = "0x00"]
    Prefix,
    #[opcode = "0x0*"]
    SameAsPrefix,
    #[opcode = "0x00_11"]
    TwoBytes,
    #[opcode = "0x**_00"]
    TwoBytesWithStars,
}

fn main() {}
