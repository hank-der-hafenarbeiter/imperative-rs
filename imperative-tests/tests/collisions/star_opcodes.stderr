error: Opcode collides with other opdcode
 --> $DIR/star_opcodes.rs:7:16
  |
7 |     #[opcode = "0x0*"]
  |                ^^^^^^

error: Collides with this opcode
 --> $DIR/star_opcodes.rs:5:16
  |
5 |     #[opcode = "0x00"]
  |                ^^^^^^
